lib/cocoapods-core/source.rb

```diff
@@ -464,6 +466,49 @@ def unchanged_github_repo?
       !GitHub.modified_since_commit(url, git_commit_hash)
     end
 
+    # @!group Sparse repo
+    #-------------------------------------------------------------------------#
+
+    def sparse_checkout_config_path
+      repo.join('.git/info/sparse-checkout')
+    end
+
+    def sparse_checkout?
+      if @is_sparse_checkout.nil?
+        @is_sparse_checkout = File.exist?(sparse_checkout_config_path)
+      end
+      @is_sparse_checkout
+    end
+
+    def sparse_checkout_file
```

Comment:
nit: this method doesn't return a file, but rather a list of strings, the name should reflect that

Reply:
done!